src/urdf-parser.cpp

```diff
@@ -223,6 +223,7 @@ namespace viewer {
         osgVector3 scale (1,1,1);
         toFloats (mesh.attribute("scale"), scale);
         meshNode->setScale(scale);
+        meshNode->applyScale();
```

Comment:
Why applyScale is not contained in setScale?

Reply:
`applyScale` modifies the underlying geometry to apply the scale and reset the variable scale to 1. For instance:
```python
node->setScale(osgVector3(0.1, 0.1, 10));
node->applyScale();
scale = node->getScale(); // contains (1., 1., 1.)
```
Here the geometry inside the node has been modified (the points of your geometry).
While this
```python
node->setScale(osgVector3(0.1, 0.1, 10));
scale = node->getScale(); // contains (0.1, 0.1, 10)
```
does not modify the geometry inside.

The name `applyScale` may be misleading. In all cases, the scale is applied. Maybe `applyScaleToGeometry` or `setGeometryScale` is better ?